tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py

```diff
@@ -0,0 +1,270 @@
+
+
+class TextPredictionV1Model(AbstractModel):
+    def __init__(self, path: str, name: str, problem_type: str,
+                 eval_metric: Optional[Union[str, metrics.Scorer]] = None,
+                 num_classes=None,
+                 stopping_metric: Optional[Union[str, metrics.Scorer]] = None,
+                 model=None,
+                 hyperparameters=None, features=None,
+                 feature_metadata: FeatureMetadata = None,
+                 debug=0, **kwargs):
+        """The TextPredictionV1Model.
+
+        The features can be a mix of
+        - text column
+        - categorical column
+        - numerical column
+
+        The labels can be categorical or numerical.
+
+        Parameters
+        ----------
+        path
+            The directory to store the modeling outputs.
+        name
+            Name of subdirectory inside path where model will be saved.
+        problem_type
+            Type of problem that this model will handle.
+            Valid options: ['binary', 'multiclass', 'regression'].
+        eval_metric
+            The evaluation metric.
+        num_classes
+            The number of classes.
+        stopping_metric
+            The stopping metric.
+        model
+            The internal model object.
+        hyperparameters
+            The hyperparameters of the model
+        features
+            Names of the features.
+        feature_metadata
+            The feature metadata.
+        debug
+            Whether to turn on debug mode
+        kwargs
+            Other arguments
+        """
+        super().__init__(path=path, name=name, problem_type=problem_type, eval_metric=eval_metric,
+                         num_classes=num_classes, stopping_metric=stopping_metric, model=model,
+                         hyperparameters=hyperparameters, features=features,
+                         feature_metadata=feature_metadata, debug=debug, **kwargs)
+        self._label_column_name = None
+
+    def _build_model(self, X_train, y_train, X_val, y_val, hyperparameters):
+        try:
+            from autogluon.text.text_prediction.text_prediction \
+                import ag_text_prediction_params, merge_params, get_column_properties, \
+                infer_problem_type, infer_eval_stop_log_metrics
+            from autogluon.text.text_prediction.models.basic_v1 import BertForTextPredictionBasic
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+
+        # Decide the name of the label column
+        if 'label' in X_train.columns:
+            label_col_id = 0
+            while True:
+                self._label_column_name = 'label{}'.format(label_col_id)
+                if self._label_column_name not in X_train.columns:
+                    break
+                label_col_id += 1
+        else:
+            self._label_column_name = 'label'
+        feature_column_properties = get_column_properties(
```

Comment:
consider using self.feature_metadata here, it already contains the type information (and additional metadata information)